src/shared/components/query/QueryStore.ts

```diff
@@ -275,6 +275,11 @@ export class QueryStore
 	@observable searchText:string = '';
 
 	@observable private _selectedStudyIds:ObservableMap<boolean> = observable.map<boolean>();
+
+	@computed get allSelectedStudyIds():string[] {

```

Comment:
Is there a reason we couldn't use the selectedStudyIds below?  Do we understand what this selectable studies thing is about?   We might take this opportunity to clean this up, or add some comments explaining why we need both of these.

Reply:
The `selectedStudyIds` uses `selectableStudiesSet` which isn't always available. It could be changed into a readonly and do an await, but would require more rewriting. Since a lot of the properties would then have to go from `@computed` to `readonly`. Not sure how much effort that would be

Reply:
to be honest, I don't actually remember why I did things that way. I don't think its a good approach. A better system would, I think, be to just delete studies from the selected study ids when we load all the studies and can see which ones are not available. This filtering system doesn't really make sense, I think, and is causing all these conceptual and coding difficulties.

Reply:
@adamabeshouse @alisman I've created a new issue for this as discussed so we can at least include this fix for a critical bug: https://github.com/cBioPortal/cbioportal/issues/3771